be/src/exec/spill/operator_mem_resource_manager.h

```diff
@@ -38,6 +44,16 @@ class OperatorMemoryResourceManager {
 
     void to_low_memory_mode();
 
+    void set_releasing() { _release_state = MEM_RELEASE_STATE::RELEASING; }
+
+    void set_release_done();
+
+    bool is_releasing() const { return _release_state == MEM_RELEASE_STATE::RELEASING; }
+
+    bool release_done() const { return _release_state == MEM_RELEASE_STATE::RELEASE_DONE; }
```

Comment:
we don't have to introduce a state for release_done/is_releasing.

Reply:
We just need to know that the operator needs to free memory

Reply:
> We just need to know that the operator needs to free memory

that's not enough, releasing buffer may take some time, we need to wait until all running releasing operations finish